amen/audio.py

```diff
@@ -49,16 +55,36 @@ def __init__(self, file_path, convert_to_mono=False, sample_rate=22050):
         ------
         An Audio object
         """
+        if file_path:
+            y, sr = librosa.load(file_path, mono=convert_to_mono, sr=sample_rate)
+        elif raw_samples is not None:
+            y = raw_samples
+            sr = sample_rate
 
         self.file_path = file_path
-        y, sr = librosa.load(file_path, mono=convert_to_mono, sr=sample_rate)
         self.sample_rate = float(sr)
+        self.analysis_sample_rate = float(analysis_sample_rate)
+
         self.raw_samples = y
         self.num_channels = y.ndim
         self.duration = librosa.get_duration(y=y, sr=sr)
+
+        if y.ndim == 2:
```

Comment:
librosa 0.4.2 will handle multi-channel resampling properly, so this block will not be necessary.